src/utils/sentry.rs

```diff
@@ -22,21 +23,68 @@ where
     }
 }
 
+impl<F> std::future::Future for SentryFuture<F>
+where
+    F: futures::future::Future,
+{
+    type Output = F::Output;
+
+    fn poll(
+        self: std::pin::Pin<&mut Self>,
+        ctx: &mut std::task::Context,
+    ) -> std::task::Poll<Self::Output> {
+        let hub = self.hub.clone();
+        // https://doc.rust-lang.org/std/pin/index.html#pinning-is-structural-for-field
+        let future = unsafe { self.map_unchecked_mut(|s| &mut s.inner) };
+        Hub::run(hub, || future.poll(ctx))
+    }
+}
+
 pub trait SentryFutureExt: Sized {
-    fn sentry_hub(self, hub: Arc<Hub>) -> SentryFuture<Self> {
-        SentryFuture { inner: self, hub }
+    fn bind_hub<H>(self, hub: H) -> SentryFuture<Self>
+    where
+        H: Into<Arc<Hub>>,
+    {
+        SentryFuture {
+            inner: self,
+            hub: hub.into(),
+        }
+    }
+
+    fn sentry_hub_current(self) -> SentryFuture<Self> {
```

Comment:
do you think these make sense to have in sentry itself?

Reply:
These were the ones that @jan-auer wanted to drop, and I think I agree.  A single `.bind_hub()` with explicitly passing in the hub you want is probably more readable.  But good point, there's no point in providing these methods on the new-and-temporary `SentryStdFutureExt`.

Reply:
Yeah, let's back those out in a future PR. We might want to add a convenience method for `Hub::new_from_top(Hub::current())` though.